docs/configuration/build_backend.md

```diff
@@ -0,0 +1,82 @@
+# The uv build backend
+
+!!! note
```

Comment:
We should also link to the build system config in here

https://docs.astral.sh/uv/concepts/projects/config/#build-systems